### ISSUE TITLE
Update the grammar for f-strings on "Lexical analysis" page

### ISSUE THREAD

**Issue by dhruvmanila** [python/cpython#116580]
# Documentation

Apologies if this is tracked somewhere else, I couldn't find it.

I'm looking at the f-string section in the "Lexical analysis" page and the grammar is not yet updated with the implementation of PEP 701: https://docs.python.org/3.12/reference/lexical_analysis.html#formatted-string-literals while it is so on the "Full Grammar specification" page: https://docs.python.org/3.13/reference/grammar.html.

The main use case for this is to link to the f-string grammar. It's difficult to do so on the "Full Grammar specification" page but it's easy on an individual section.

**Comment by terryjreedy** [comment]
@pablogsal

**Comment by hugovk** [comment]
Also cc @encukou who was been working on grammar docs.

**Comment by encukou** [comment]
Yeah, I'm giving a bit of time each week to generating docs from the actual grammar, and hope to do it by 3.13. But i want to look at f-strings (and other topics that need touch lexical docs) *last*.

**Comment by encukou** [comment]
No issue yet, just [discussion](https://discuss.python.org/t//36709).
I'll assign this to myself: it'll be a few months before *I* get to this, but do talk to me if you want to start working on it earlier :)

**Comment by encukou** [comment]
See https://github.com/python/cpython/pull/137469
